api/models/service.go

```diff
@@ -73,19 +74,24 @@ func NewService(model *Model, version *Version, modelOpt *ModelOption, endpoint
 			Team:      model.Project.Team,
 			Labels:    MergeProjectVersionLabels(model.Project.Labels, version.Labels),
 		},
-		Transformer:       endpoint.Transformer,
-		Logger:            endpoint.Logger,
-		DeploymentMode:    endpoint.DeploymentMode,
-		AutoscalingPolicy: endpoint.AutoscalingPolicy,
-		Protocol:          endpoint.Protocol,
-		CurrentIsvcName:   endpoint.InferenceServiceName,
+		Transformer:               endpoint.Transformer,
+		Logger:                    endpoint.Logger,
+		DeploymentMode:            endpoint.DeploymentMode,
+		AutoscalingPolicy:         endpoint.AutoscalingPolicy,
+		Protocol:                  endpoint.Protocol,
+		CurrentIsvcName:           endpoint.InferenceServiceName,
+		EnabledModelObservability: endpoint.EnableModelObservability,
 	}
 }
 
 func (svc *Service) GetPredictionLogTopic() string {
 	return fmt.Sprintf("caraml-%s-%s-prediction-log", svc.Namespace, svc.ModelName)
 }
 
+func (svc *Service) GetPredictionLogTopicPerVersion() string {
```

Comment:
Nit: Can we call this `GetPredictionLogTopicForVersion` instead of `GetPredictionLogTopicPerVersion` to indicate that it is for the specific version? "Per" version makes it sound like we will return 1 topic per existing version of the model in the same API call.